src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/chatpageobject/ChatPage.java

```diff
@@ -0,0 +1,415 @@
+  private List<WebElement> adminDropDownActionsElements;
+  @FindBy(css = "#Rail h1.private")
+  private WebElement privateMessagesHeader;
+  @FindBy(css = "#Rail h1.wordmark")
+  private WebElement chatWordmarkImage;
+  @FindBy(css = "#WikiChatList li")
+  private WebElement chatLoadedIndicator;
+  @FindBy(css = "#ChatHeader h1.private")
+  private WebElement privateChatHeader;
+  @FindBy(css = "div.actions .admin-actions .give-chat-mod")
+  private WebElement addUserModStatusButton;
+  @FindBy(css = "div#UserStatsMenu.UserStatsMenu")
+  private WebElement userOptions;
+  @FindBy(css = "ul.admin-actions")
+  private WebElement userModOptions;
+  @FindBy(css = "li.message-wall")
+  private WebElement userMessageWallButton;
+  @FindBy(css = "div.speech-bubble-message")
+  private WebElement userPageMessageWallTab;
+  @FindBy(css = "li.contribs")
+  private WebElement userContributionsButton;
+  @FindBy(css = "table.mw-contributions-table")
+  private WebElement userPageContributionsTab;
+  @FindBy(css = "form#Write.Write.blocked")
+  private WebElement userBlockedMessageField;
+  @FindBy(css = "div.header-column.header-title")
+  private WebElement permissionsErrorTitle;
+  @FindBy(xpath = "//h1")
+  private WebElement userNameTitle;
+
+  private static final String USER_UNBAN_LINK = "//a[@data-type='ban-undo' and @data-user='%s']";
+  private static final String USER_UNBAN_CONFIRM_MESSAGE =
+          "//div[@class='Chat']//li[contains(text(), 'has ended the Chat ban for %s')]";
+  private static final String USER_SELECTOR = "#user-%s";
+  private static final String PRIVATE_MESSAGE_USER_SELECTOR = "#priv-user-%s";
+  private static final String PATH_MESSAGE_ON_CHAT = "//span[@class='message'][contains(text(), '%s')]";
+  private static final String NOTIFICATION_COUNTER =
+          "//span[@class='splotch' and contains(text(), '%s')]";
+
+  private static final int REGULAR_USER_DROPDOWN_ELEMENTS = 3;
+
+  public ChatPage open() {
+    getUrl(urlBuilder.getUrlForWiki() + URLsContent.SPECIAL_CHAT);
+
+    return this;
+  }
+
+  private WebElement userOnChat(String userName, String message) {
+    return driver.findElement(By.cssSelector(String.format(message, userName)));
+  }
+
+  private WebElement userPostedMessage(String message) {
+    return driver.findElement(By.xpath(String.format(PATH_MESSAGE_ON_CHAT, message)));
+  }
+
+  private WebElement userPrivateMessageNotification(int notificationCount) {
+    return driver.findElement(By.xpath(String.format(NOTIFICATION_COUNTER, notificationCount)));
+  }
+
+  private WebElement getUserUnbanMessage(String userName) {
+    return driver.findElement(By.xpath(String.format(USER_UNBAN_CONFIRM_MESSAGE, userName)));
+  }
+
+  private WebElement getUserUnbanLink(String userName) {
+   return driver.findElement(By.xpath(String.format(USER_UNBAN_LINK, userName)));
+  }
+
+  public boolean isUserOnChat() {
+    try {
+      wait.forElementVisible(chatInlineAlert);
```

Comment:
isn't `wait.forElementVisible` checks if element is displayed? if yes, we can return `true` instead of `chatInlineAlert.isDisplayed`